docs/release_guide.md

```diff
@@ -121,3 +123,28 @@ Make an announcement on the [mailing list](https://groups.google.com/g/openservi
 Skip this step if the release is a release candidate (RC).
 
 Open a pull request against the `main` branch making the same version updates as [above](#update-release-branch-with-patches-and-versioning-changes) so the latest release assets are referenced there.
+
+## Make version changes on docs.openservicemesh.io
+
+Associated version numbers need to be updated in the [docs.openservicemesh.io](https://github.com/openservicemesh/osm-docs/) repo.
+
+The files that need to be updated are:
```

Comment:
I only looked at the first few files here, but several seemed like they don't need to be included. Please take another look at these to trim it down as much as possible.

Reply:
I further trimmed the list down. The criteria is, we only update the docs that version number affects sample shell commands, or links to the latest code.

Reply:
Thanks for doing that, I think some should be added back though. I'm going through https://github.com/openservicemesh/osm-docs/pull/109 to see if there's anything missing there. That PR changed a few spots that we don't need to update every time, but I'll leave comments on each of those spots. Then let's make this list based off of the changes there. I'll let you know when I'm finished reviewing that PR.

Sorry this is turning into a bigger deal than it probably should, but making this list complete and concise should save us headaches down the road.

Reply:
@nojnhuh I think osm-docs #109 is finished. I will merge this PR for v0.9.x release

Reply:
Is it worth including a list here? Or should we guide devs to just use the search function?

Reply:
I actually think so. We do the same, and the list is not always accurate.

Reply:
@nojnhuh I removed the concrete files list here.